pkg/gardenlet/operation/botanist/cleanup.go

```diff
@@ -138,7 +139,7 @@ func cleanResourceFn(cleanOps utilclient.CleanOps, c client.Client, list client.
 				if utilclient.AreObjectsRemaining(err) {
 					return retry.MinorError(helper.NewErrorWithCodes(err, gardencorev1beta1.ErrorCleanupClusterResources))
 				}
-				return retry.SevereError(err)
+				return retry.SevereError(fmt.Errorf("failed cleanup of resource kind %s : %w", list.GetObjectKind(), err))
```

Comment:
Cosmetic suggestion:

```suggestion
				return retry.SevereError(fmt.Errorf("failed to clean up resources with kind %s: %w", list.GetObjectKind(), err))
```